automaton/spike.go

```diff
@@ -1,8 +1,104 @@
 package automaton
 
+import (
+	"github.com/R-jim/Momentum/aggregate/carrier"
+	"github.com/R-jim/Momentum/aggregate/knight"
+	"github.com/R-jim/Momentum/aggregate/spike"
+	"github.com/R-jim/Momentum/operator"
+	"github.com/R-jim/Momentum/util"
+)
+
 /*
 TODO:
-spike will patrol around the artifact in slow pace, next closest position to artifact
-if detect carrier/knight will switch to multiplier mode and move in to intercept
 attack carrier/knight when in range 1 from spike
 */
+
+type SpikeAutomaton interface {
+	Auto(id string) error
+}
+
+type spikeImpl struct {
+	carrierStore carrier.Store
+	knightStore  knight.Store
+	spikeStore   spike.Store
+
+	operator operator.Operator
+}
+
+func NewSpikeAutomaton(carrierStore carrier.Store, knightStore knight.Store, spikeStore spike.Store, operator operator.Operator) SpikeAutomaton {
+	return spikeImpl{
+		carrierStore: carrierStore,
+		knightStore:  knightStore,
+		spikeStore:   spikeStore,
+
+		operator: operator,
+	}
+}
+
+func (i spikeImpl) Auto(id string) error {
+	return i.AutoStrike(id)
+}
+
+func (i spikeImpl) AutoStrike(id string) error {
+	positionState, err := spike.GetPositionState(i.spikeStore, id)
+	if err != nil {
+		return err
+	}
+
+	targetKnightIDs := []string{}
+	targetCarrierIDs := []string{}
```

Comment:
Can separate this to a different `targetIDs` store. Have an `AutoSearch` to update the store. `AutoStrike` will read from said store

Reply:
Added to [Automaton enhancements](https://github.com/R-Jim/Momentum/issues/10)